crates/bevy_ecs/src/storage/table/mod.rs

```diff
@@ -652,25 +329,65 @@ impl Table {
         }
     }
 
+    /// Writes component data to its corresponding column at the given row.
+    /// Assumes the slot is uninitialized, drop is not called.
+    /// To overwrite an existing initialized value, use [`Self::replace_component`] instead.
+    ///
+    /// # Safety
+    /// - `row.as_usize()` < `len`
+    /// - `comp_ptr` holds a component that matches the `component_id`
+    pub unsafe fn initialize_component(
+        &mut self,
```

Comment:
There's actually no way to get a `&mut Table` in the public API, so all functions here that takes a `&mut self` might be unusable in this current state.